### PR TITLE
Add resource to manage solution landing zone [1]

### PR DESCRIPTION
**Pull request by Didainius**
This is the first step towards adding support for solution add-on management in VCD using Terraform provider.
This particular PR adds one new resource and one data source `vcd_solution_landing_zone` that is responsible for configuring Solution Landing Zone (SLZ)

### CODE REVIEW

File: vcd/datasource_vcd_solution_landing_zone.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "VDC Name",
+						},
+						"is_default": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Shows is the member is enabled or not",
```
Comment (lvirbalas) [inline vcd/datasource_vcd_solution_landing_zone.go:105]: Found a typo.

**Review by lvirbalas — APPROVED** [review]
Nice!! One comment inline.

File: vcd/datasource_vcd_solution_landing_zone.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dsSlzChildComponent(title string) *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeSet,
+		Required:    true,
+		Description: "",
```
Comment (adambarreiro) [inline vcd/datasource_vcd_solution_landing_zone.go:15]: Description missing. Maybe something like `"Details of " + title` or something like that.

File: vcd/datasource_vcd_solution_landing_zone.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+						},
+						"is_default": {
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Shows is the member is enabled or not",
+						},
+						"capabilities": {
+							Type:        schema.TypeSet,
+							Computed:    true,
+							Description: "",
```
Comment (adambarreiro) [inline vcd/datasource_vcd_solution_landing_zone.go:110]: Missing description

File: website/docs/r/solution_landing_zone.html.markdown [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+* `state` - reports the state of parent [Runtime Defined
+  Entity](/providers/vmware/vcd/latest/docs/resources/rde)
+
+## Importing
+
+~> The current implementation of Terraform import can only import resources into the state.
+It does not generate configuration. [More information.](https://www.terraform.io/docs/import/)
+
+Only a single entry should exist in a valid scenario, but :
```
Comment (adambarreiro) [inline website/docs/r/solution_landing_zone.html.markdown:111]: Something is missing here I think

File: website/docs/r/solution_landing_zone.html.markdown [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+* `org_vdc_network` - (Required) At least one Org VDC Network is required. See [vdc
+  child](#vdc-child) block description for possible values.
+* `compute_policy` - (Required) At least Compute Policy is required. See [vdc child](#vdc-child)
+  block description for possible values.
+* `storage_policy` - (Required) At least Storage Policy is required. See [vdc child](#vdc-child)
+  block description for possible values.
+
+
+<a id="vdc-child"></a>
+## VDC child configuration block
```
Comment (adambarreiro) [inline website/docs/r/solution_landing_zone.html.markdown:93]: Should this just be "Child configuration block"? Given that it's used for all three blocks.

**Review by adambarreiro — COMMENTED** [review]
First scan

File: vcd/datasource_vcd_solution_landing_zone.go (hunk 1 above)
Comment (Didainius) [reply vcd/datasource_vcd_solution_landing_zone.go:105]: Fixed.

File: .changes/v3.13.0/1251-features.md [hunk 6]
```diff
@@ -0,0 +1,2 @@
+* **New Resource:** `vcd_solution_landing_zone` to manage solution add-on landing zones [GH-1251]
+* **New Data Source:** `vcd_solution_landing_zone` to read solution add-on landing zones [GH-1251]
```
[the remark this answers is not in the document]
Comment (Didainius) [reply .changes/v3.13.0/1251-features.md:2]: True, single one. Updated.

File: vcd/datasource_vcd_solution_landing_zone.go (hunk 2 above)
Comment (Didainius) [reply vcd/datasource_vcd_solution_landing_zone.go:15]: Indeed. Fixed.

File: vcd/datasource_vcd_solution_landing_zone.go (hunk 3 above)
Comment (Didainius) [reply vcd/datasource_vcd_solution_landing_zone.go:110]: Fixed

File: website/docs/r/solution_landing_zone.html.markdown (hunk 4 above)
Comment (Didainius) [reply website/docs/r/solution_landing_zone.html.markdown:111]: True, I messed this up with next resource in mind.

File: website/docs/r/solution_landing_zone.html.markdown (hunk 5 above)
Comment (Didainius) [reply website/docs/r/solution_landing_zone.html.markdown:93]: Changed - have a look if this reads clearer now. I had my doubts for this level of nesting :)

**Review by dataclouder — APPROVED** [review]

**Review by adambarreiro — APPROVED** [review]

**Pull request merged**